library/src/main/java/com/google/maps/android/data/kml/KmlLayer.java

```diff
@@ -106,14 +120,48 @@ public KmlLayer(GoogleMap map, InputStream stream, FragmentActivity activity, Ma
         if (stream == null) {
             throw new IllegalArgumentException("KML InputStream cannot be null");
         }
-        KmlRenderer mRenderer = new KmlRenderer(map, activity, markerManager, polygonManager, polylineManager, groundOverlayManager);
+        KmlRenderer renderer = new KmlRenderer(map, activity, markerManager, polygonManager, polylineManager, groundOverlayManager);
+
+        BufferedInputStream bis = new BufferedInputStream(stream);
+        bis.mark(1024);
+        ZipInputStream zip = new ZipInputStream(bis);
+        try {
+            KmlParser parser = null;
+            ZipEntry entry = zip.getNextEntry();
+            if (entry != null) { // is a KMZ zip file
+                while (entry != null) {
+                    if (parser == null && entry.getName().toLowerCase().endsWith(".kml")) {
+                        parser = parseKml(zip);
+                    } else {
+                        Bitmap bitmap = BitmapFactory.decodeStream(zip);
```

Comment:
Since other file types might also be used (e.g. `.mp3`), it would be good to check that an image is used and if not, log via `Log.w` that the file type is not supported.

Reply:
Will this handle subdirectories?

```
sample.kmz
 - doc.kml
 - image.png
 - dir/image2.png
 - dir/image3.png
 - dir/dir2/image4.png
 ...
```

Reply:
Yes, subdirectories are handled. The name includes the full file path within the zip archive.

Reply:
Added warning log: https://github.com/googlemaps/android-maps-utils/pull/625/commits/3d1d0e7c7f46ec08e3829ddb8a7e44e420ff6ff2